openwisp_monitoring/views.py

```diff
@@ -32,8 +33,49 @@ def _get_additional_data(request, *args, **kwargs):
         """
         return {}
 
+    def _add_custom_date_group_map(self, time):
+        """
+        Adds custom date range time
+        grouping to chart.GROUP_MAP
+        """
+        # If custom date time grouping already exists
+        # with default time group then pop that time
+        if len(Chart.GROUP_MAP.items()) > 5:
+            Chart.GROUP_MAP.popitem()
+        group = '10m'
+        days = int(time.split('d')[0])
+        # custom grouping between 1 to 2 days
+        if days > 0 and days < 3:
+            group = '10m'
+        # custom grouping between 3 to 6 days (base 5)
+        elif days >= 3 and days < 7:
+            group = str(5 * round(((days / 3) * 20) / 5)) + 'm'
+        # custom grouping between 8 to 27 days
+        elif days > 7 and days < 28:
+            group = str(round(days / 7)) + 'h'
+        # custom grouping between 28 to 364 days
+        elif days >= 28 and days < 365:
+            group = str(round(days / 28)) + 'd'
+        Chart.GROUP_MAP.update({time: group})
```

Comment:
It doesn't feel correct to update a model constant like this in the view. I will look for alternative ways to achieve this functionality. For now, it is good.

Reply:
I'll also look for alternatives for this. :+1:

Reply:
Hey, let's not modify a global data structure, rather we should work on a copy of it and return that for usage whenever we need to modify it, or the global datastructure may become huge since it's modified by many requests and it can yield unpredictable results.

Reply:
ok cool :+1: